hotsos/core/plugintools.py

```diff
@@ -224,6 +247,113 @@ def bind_interfaces(self):
         """
 
 
+def summary_entry(name, index=0):
+    """Decorate a class member function to indicate that it is a summary
+    function. Summary functions are automatically discovered by the
+    PluginPartBase class and will be part of the summary output."""
+
+    def real_decorator(func):
+        # Store the summary metadata information on the function itself.
+        func.__summary_elem_name = name
+        func.__summary_elem_index = index
+        return func
+    return real_decorator
+
+
+def get_min_available_entry_index():
+    """ Return first index available after default entries. """
+    return DefaultSummaryEntryIndexes.available
+
+
+@dataclass(frozen=True)
+class DefaultSummaryEntryIndexes():
```

Comment:
I think what you need here is [an enum](https://docs.python.org/3/library/enum.html) instead of a dataclass:

```python

from enum import IntEnum, auto

class DefaultSummaryEntryIndexes(IntEnum):
	version = 0
	release = auto()
	services = auto()
	snaps = auto()
	dpkg = auto()
	docker_images = auto()
	last = auto()


print(f"{list(DefaultSummaryEntryIndexes)}")

# [<DefaultSummaryEntryIndexes.version: 0>, <DefaultSummaryEntryIndexes.release: 1>, <DefaultSummaryEntryIndexes.services: 2>, <DefaultSummaryEntryIndexes.snaps: 3>, <DefaultSummaryEntryIndexes.dpkg: 4>, <DefaultSummaryEntryIndexes.docker_images: 5>, <DefaultSummaryEntryIndexes.last: 6>]

print(f"{8 + DefaultSummaryEntryIndexes.last}")
# 14
```

Enum members are immutable by default.

Reply:
true, im' drinking the dataclass koolaid too much ;)

Reply:
fixed.